app/controllers/course/material/folders_controller.rb

```diff
@@ -1,30 +1,32 @@
 # frozen_string_literal: true
 class Course::Material::FoldersController < Course::Material::Controller
+  rescue_from ActiveRecord::RecordNotFound, with: :handle_not_found
+  skip_load_resource :folder, only: [:index]
   before_action :authorize_read_owner!, only: [:show, :download]
 
+  def index
+    @folder = current_course.root_folder
+    load_subfolders
+    render 'show'
+  rescue ActiveRecord::RecordNotFound
```

Comment:
1. In normal application workflow, it's not possible for `root_folder` to not exist. However, it's possible to call destroy on the root folder, as well as make changes directly in the database.
2. The `rescue_from` calls `handle_not_found` which will query for the root folder. No point querying for the `root_folder` again in `handle_not_found` if I already know it can't be found in `#index`.

Reply:
1. All those near-impossible scenario that you enlisted there might only be caused by one of us developers somehow screw things up with our DB and we somehow let loose of the permission to destroy folder. I'm not sure if this is necessary, but for the sake of handling edge-cases, it should be fine.
2. Got it. So what you want to achieve here is that should any folder-related operations throw an error due to `RecordNotFound`, then the `index` controller operation will take place. With the exception if that mentioned operation is `index`, then just throw an error

Reply:
as per discussed, should be fine to do exception handling for the case of root folder not found